File: src/main/java/com/rinhajava/app/domain/Pessoa.java
package com.rinhajava.app.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.UUID;

@Data
public class Pessoa {

    @JsonProperty("id")
    private UUID id;

    @JsonProperty("apelido")
    private String apelido;

    @JsonProperty("nome")
    private String nome;

    @JsonProperty("nascimento")
    private String nascimento;

    @JsonProperty("stack")
    private String[] stack;

    public Pessoa() {

    }

    public Pessoa(UUID id, String apelido, String nome, String nascimento, String[] stack) {
        this.id = id;
        this.apelido = apelido;
        this.nome = nome;
        this.nascimento = nascimento;
        this.stack = stack;
    }
}

File: src/model/pessoa.go
package model


type Pessoa struct {
    Id          string `json:"id"`
    Apelido     string `json:"apelido"`
    Nome        string `json:"nome"`
    Nascimento  string `json:"nascimento"`
    Stack       []string `json:"stack"`
}

type PessoaCadastro struct {
    Id          string `json:"id"`
    Apelido     string `json:"apelido"`
    Nome        string `json:"nome"`
    Nascimento  string `json:"nascimento"`
    Stack       string `json:"stack"`
    CampoBusca  string
}
